fmn/core/constants.py

```diff
@@ -1,7 +1,18 @@
-from enum import Enum
+from enum import Enum, EnumMeta
+from functools import cache
 
 
-class ArtifactType(Enum):
+class ContainsEnumMeta(EnumMeta):
+    @cache  # noqa: B019
```

Comment:
Are you sure that this can't leak? It does not seem to be a singleton this time.

Reply:
It’s the metaclass used for `ArtifactType`, so effectively a singleton (`cls` will always be `ArtifactType`). I’m not super happy with the implementation, should I make this a separate class method of `ArtifactType` instead and use it explicitly?

Reply:
Yeah, maybe something like `has_name` and `has_value` would also make it clearer what we're testing here. Of course you only need the latter so no need to implement the other one ;-)

Reply:
Heh, something like `has_name()` already exists even though it might not be well-known, e.g.: `"packages" in ArtifactType.__members__`